Floofbot/Services/EventLoggerService.cs

```diff
@@ -0,0 +1,496 @@
+
+namespace Floofbot.Services
+{
+    public class EventLoggerService
+    {
+
+        private FloofDataContext _floofDb;
+        private DiscordSocketClient _client;
+        public EventLoggerService(FloofDataContext floofDb, DiscordSocketClient client)
+        {
+
+            _floofDb = floofDb;
+            _client = client;
+            _client.MessageUpdated += MessageUpdated;
+            _client.MessageDeleted += MessageDeleted;
+            _client.UserBanned += UserBanned;
+            _client.UserUnbanned += UserUnbanned;
+            _client.UserJoined +=UserJoined;
+            _client.UserLeft += UserLeft;
+            _client.GuildMemberUpdated += GuildMemberUpdated;
+        }
+        public async Task<ITextChannel> GetChannel(Discord.IGuild guild, string eventName = null)
+        {
+            if (eventName == null)
+                return null;
+
+            var serverConfig = _floofDb.LogConfigs.Find(guild.Id);
+            System.Reflection.PropertyInfo propertyInfo = serverConfig.GetType().GetProperty(eventName);
+            ulong logChannel = (ulong)(propertyInfo.GetValue(serverConfig, null));
+            var textChannel = await guild.GetTextChannelAsync(logChannel);
+            return textChannel;
+        }
+        public bool IsToggled(IGuild guild)
+        {
+            // check if the logger is toggled on in this server
+            // check the status of logger
+            var ServerConfig = _floofDb.LogConfigs.Find(guild.Id);
+            if (ServerConfig == null) // no entry in DB for server - not configured
+                return false;
+
+            return ServerConfig.IsOn;
+        }
+
+        public async Task MessageUpdated(Cacheable<IMessage, ulong> before, SocketMessage after, ISocketMessageChannel chan)
+        {
+            try
+            {
+                // deal with empty message
+                var messageBefore = (before.HasValue ? before.Value : null) as IUserMessage;
+                if (messageBefore == null)
+                    return;
+
+                var channel = chan as ITextChannel; // channel null, dm message?
+                if (channel == null)
+                    return;
+
+                if (messageBefore.Content == after.Content) // no change
+                    return;
+
+                if ((IsToggled(channel.Guild)) == false) // not toggled on
+                    return;
+
+                Discord.ITextChannel logChannel = await GetChannel(channel.Guild, "MessageEditedChannel");
+                if (channel == null)
+                    return;
+
+                var embed = new EmbedBuilder()
+                 .WithTitle($"⚠️ Message Edited | {after.Author.Username}")
+                 .WithColor(Color.DarkGrey)
+                 .WithDescription($"{after.Author.Mention} ({after.Author.Id}) has edited their message in {channel.Mention}!")
+                 .AddField("Before", messageBefore.Content)
+                 .AddField("After", after.Content)
```

Comment:
For these fields, I'd suggest making them [inline](https://discord.foxbot.me/docs/api/Discord.EmbedBuilder.html?q=EmbedBuilder#Discord_EmbedBuilder_AddField_System_String_System_Object_System_Boolean_) to minimize the size of embeds. I'd also argue that we should remove the profile picture display from log events that don't have much to do with the profile picture, but in this case I don't think it actually changes the display length of the embed.

Reply:
done